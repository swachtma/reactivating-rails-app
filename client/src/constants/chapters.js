export const LOAD_CHAPTERS = "LOAD_CHAPTERS";