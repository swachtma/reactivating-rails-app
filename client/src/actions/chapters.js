import * as constants from '../constants/chapters'

export const loadChapters = (payload) => ({ type: constants.LOAD_CHAPTERS, payload })
