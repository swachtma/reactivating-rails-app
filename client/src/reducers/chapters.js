import * as constants from '../constants/chapters';

let default_state = [];

export const sortChapters = (chapters) => {
  return chapters.sort((a,b) =>{
    return a.id - b.id;
  });
};

const chaptersReducer = function(state = default_state,action){
  switch (action.type) {
    case(constants.LOAD_CHAPTERS):
      return [...sortChapters(action.payload)];
    default:
      return state;
  }
};

export default chaptersReducer;