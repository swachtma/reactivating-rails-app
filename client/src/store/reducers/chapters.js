import * as constants from '../../constants/chapters';

let default_state = [];

export const sortChapters = (chapters) => 
  chapters.sort((a,b) => a.id - b.id);

export default function chaptersReducer(state = default_state,action){
  switch (action.type) {
    case(constants.LOAD_CHAPTERS):
      return [...sortChapters(action.payload)];
    default:
      return state;
  }
}