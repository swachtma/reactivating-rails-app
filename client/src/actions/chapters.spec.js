/* global expect */
import * as actions from './chapters'

describe('Chapters actionCreators', () => {
  test('loadChapters(chapters)', () => {
    expect(actions.loadChapters([{ contents: 'chapter' }])).toMatchSnapshot()
  })
})
