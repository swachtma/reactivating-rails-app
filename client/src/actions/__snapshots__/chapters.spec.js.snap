// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chapters actionCreators loadChapters(chapters) 1`] = `
Object {
  "payload": Array [
    Object {
      "contents": "chapter",
    },
  ],
  "type": "LOAD_CHAPTERS",
}
`;
